### DISCUSSION TITLE
將 Blazor WebAssembly 整合進 Tauri 應用程式 | 全端開發人員天梯

### DISCUSSION THREAD

**Discussion by giscus** [wellwind/wellwind.github.io#79] [Announcements]
# 將 Blazor WebAssembly 整合進 Tauri 應用程式 | 全端開發人員天梯

Tauri 內建了很多前端的專案範本，但可惜的是沒有 Blazor WebAssembly 的範本，但 Tauri 架構本身已經考量得很完善了，只要最終產出是純前端的網站，任何應用程式都可以包裝到 Tauri 內，今天這篇文章就來說明一下如何將 Blazor WebAssembly 加到 Tauri 應用程式內。

https://fullstackladder.dev/blog/2022/12/10/tauri-with-blazor-web-assembly/

**Comment by Seanwanq** [comment]
您好，我读了您的帖子，受益很多，但是在调用系统api的时候遇到了一些问题。我希望可以自定义放大缩小窗口按钮，用的代码是

```c#
await JS.InvokeVoidAsync(
            "window.__TAURI__.window.toggleMaximize",
            );
```

但是应用一直显示报错，理由是无法找到相应的函数。请问该如何正确地调用相关的函数呢？

**Reply by wellwind** [reply]
我猜可能是因為你沒有在 `tauri.conf.json` 內設定權限
https://tauri.app/v1/api/js/window

**Reply by Seanwanq** [reply]
谢谢，我已经明白了，需要写成
```c#
await JS.InvokeVoidAsync(
        "window.__TAURI__.window.appWindow.toggleMaximize"
        );
```
才行。您启发了我。
